Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -1100,34 +1100,37 @@ def checkout(self, argument, prune=None, prompt=False):
 
         # Branch is not dev and exists on a remote.
         if not match and branch_remote:
-            remote_path = '{}/{}'.format(branch_remote, argument)
-            local_head = self.commit(branch=argument, include_log=False, include_identifier=False)
-            remote_head = self.commit(branch=remote_path, include_log=False, include_identifier=False)
-            local_bp = self.branch_point(ref=local_head.hash)
-            merge_base_with_target_remote = run(
-                [self.executable(), 'merge-base', local_bp.hash, remote_head.hash],
-                cwd=self.root_path,
-                capture_output=True,
-                encoding='utf-8',
-            ).stdout.strip()
-
-            # Resets branch if local is not tracking force pushed remote.
-            if merge_base_with_target_remote != local_bp.hash:
-                if local_bp.hash != local_head.hash:
-                    log.info(" You have unsaved changes on the local branch.")
-                    if prompt and Terminal.choose(
-                        "Local changes on {} will not be saved. Would you like to override the local version of this branch with the version from '{}'?".format(argument, path),
-                        default='No'
-                    ) == 'No':
-                        sys.stderr.write("Checkout aborted.\n")
-                        return None
-                log.info(" Resetting branch {} to remote {}. Checkout will erase all local changes on {}.\n".format(argument, remote_path, argument))
-                return None if run(
-                    [self.executable(), 'checkout'] + ['-B', argument, remote_path] + log_arg,
+            try:
+                remote_path = '{}/{}'.format(branch_remote, argument)
+                local_head = self.commit(branch=argument, include_log=False, include_identifier=False)
```

Comment:
Can we make this `try/catch` much smaller? I'm pretty sure it's just this line right here that needs it.

In general, `try/catch` in Python is a bit dangerous to apply this broadly because it can, for example, actually surpress syntax errors.